### ISSUE TITLE
Keras: model.fit_generator or model.fit not working properly

### ISSUE THREAD

**Issue by sanidhyamangal** [tensorflow/tensorflow#38938]
<em>Please make sure that this is a bug. As per our
[GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md),
we only address code/doc bugs, performance issues, feature requests and
build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution: Ubuntu 18.04.3 LTS
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.2.0-rc3
- Python version: 3.6.9
- CUDA/cuDNN version: (colab)
- GPU model and memory: (colab)

**Describe the current behavior**
While using ```model.fit``` or ```model.fit_generator``` the sub iterations in epoch shows unknown number of iterations which is goes on despite surpassing the batch size.

**Describe the expected behavior**
While using ```model.fit``` or ```model.fit_generator``` the sub iterations in epoch must be definite.

**Standalone code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate
the problem. If possible, please share a link to [Colab](https://colab.research.google.com/drive/1dWbszUUjRagXyJUUqjdsM1yXtaAmKc23).

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.

```shell
WARNING:tensorflow:From <ipython-input-23-2da3481afcb0>:1: Model.fit_generator (from tensorflow.python.keras.engine.training) is deprecated and will be removed in a future version.
Instructions for updating:
Please use Model.fit, which supports generators.
Epoch 1/20
     13/Unknown - 20s 2s/step - loss: 1.0732 - accuracy: 0.2602
```

**Comment by ravikyram** [comment]
@sanidhyamangal 

I am not able to access the colab link you have provided. Please, provide access so i can try to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by sanidhyamangal** [comment]
@ravikyram, I have provided you the access, kindly check.

**Comment by jetjodh** [comment]
I checked the colab notebook and was able to reproduce the issue. For now, you may use tf -2.1 which does not contain this issue.

**Comment by sanidhyamangal** [comment]
@jetjodh, I already rolled back to previous iterations. Just reporting the bug so that it doesn’t appear in 2.2.0 stable release.

**Comment by ravikyram** [comment]
I am not seeing any issue with TF 2.1.0 .However i am able to reproduce the issue with TF 2.2.0-rc3. Please, find the gist [here.](https://colab.sandbox.google.com/gist/ravikyram/eeee97ce61ffb4fbf653ec375fe97fe8/untitled834.ipynb).Thanks!

**Comment by sanidhyamangal** [comment]
@ravikyram, I have already used migrated my code from 2.2.0-rc3 to 2.1.0 and 2.0.0. Thanks for gist. Hoping to see this issue resolved in upcoming stable build.

**Comment by goldiegadde** [comment]
@sanidhyamangal thanks for reporting the issue. this issue is fixed in the latest tf-nightly and will be available with the upcoming 2.3.0 release.

**Comment by ravikyram** [comment]
@sanidhyamangal 

This was also resolved in TF nightly version(`2.3.0-dev20200617`). Please check the gist [here.](https://colab.research.google.com/gist/ravikyram/86d88c05a705c79b7ee1df625ef7ccad/untitled44.ipynb).You could use tf-nightly for now and in the next couple of months new stable version will be released.Please verify once and close the issue. Thanks!

**Comment by sanidhyamangal** [comment]
@ravikyram 

Yes, this one is resolved and I am no longer able to reproduce this issue in new version. Thanks for resolving this.
